### PR TITLE
UPSTREAM: <carry>: openshift: describe how to deploy machine API with azure machine controller over kubernetes

### PR DESCRIPTION
**Pull request by ingvagabund**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: config/bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+######## Deploy kubernetes master
+################################################
+
+kubeadm init --apiserver-bind-port 8443 --token 2iqzqm.85bs0x6miyx1nm7l --apiserver-cert-extra-sans=$(curl icanhazip.com) --pod-network-cidr=192.168.0.0/16 -v 6
+
+# Enable networking by default.
+kubectl apply -f https://raw.githubusercontent.com/cloudnativelabs/kube-router/master/daemonset/kubeadm-kuberouter.yaml --kubeconfig /etc/kubernetes/admin.conf
+
+# Binaries expected under /opt/cni/bin are actually under /usr/libexec/cni
+mkdir -p /opt/cni/bin
```
Comment (frobware) [inline config/bootstrap.sh]: I think these might be simpler as:

`cp /usr/libexec/cni/<foo> /opt/cni/bin`

Repeat...

File: config/bootstrap.sh [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+mkdir -p /root/.kube
+cp -i /etc/kubernetes/admin.conf /root/.kube/config
+chown $(id -u):$(id -g) /root/.kube/config
+
+################################################
+######## Deploy machine-api plane
+################################################
+
+git clone https://github.com/openshift/cluster-api-provider-azure.git
```
Comment (frobware) [inline config/bootstrap.sh]: Maybe consider checking out the branch as part of the clone?

`git clone -b how-to-deploy-azure-actuator https://...`

File: config/generate-bootstrap.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+if [ -z "$AZURE_SUBSCRIPTION_ID" ]; then
```
Comment (frobware) [inline config/generate-bootstrap.sh]: An alternative form and shorter form is:

```
set -eux
: ${AZURE_SUBSCRIPTION_ID:?not set}
```

File: config/bootstrap.sh (hunk 1 above)
Comment (ingvagabund) [reply config/bootstrap.sh]: done

File: config/bootstrap.sh (hunk 2 above)
Comment (ingvagabund) [reply config/bootstrap.sh]: No longer in need to checkout another branch. All the manifests are now part of the master branch,

**Pull request merged**